### ISSUE TITLE
accessibility: "More Dates" popup on Course Pages lacks keyboard controls

### ISSUE THREAD

**Issue by pdpinch** [mitodl/mitxonline#1207]
### Steps to Reproduce

1. go to any product page with multiple runs
2. Click on the dates to display the pop-up

### Expected Behavior

- should be able to tab between the dates, because they are links

### Actual Behavior

- can't tab

### Related Issues

- #122 

### Screenshot or Screencast

![image](https://github.mit.edu/storage/user/18337/files/788ce400-4a28-11ed-9dc9-5fae3c927680)

**Comment by pdpinch** [comment]
> See https://www.w3.org/WAI/ARIA/apg/example-index/link/link.html for missing code elements/examples.
Need role=link and onkeydown=

**Comment by ChristopherChudzicki** [comment]
@collinpreston I looked into this a while ago but never actually managed to view the popover in a browser. (I was only trying to view it on rc. IIRC, We discovered that in order for "more dates" to show, you need a previous enrollment, which I did not have.)

One thing I noticed in the code is that the contents are not focusable (I believe; like I said, I couldn't actually try it). They are spans but should be anchor tags.

Additionally... I am not sure what the default container for bootstrap popovers is. Appending the popovers to the parent element may help with tab navigation (once the elements are focusable), but possibly could mess up scrolling.

**Comment by rachellougee** [comment]
When I looked at it last week, the problem appeared to be the popover is not focusable as Chris mentioned above. I came across this https://getbootstrap.com/docs/4.4/components/popovers/#usage document from bootstrap, It mentioned the tabbing issue for the popover. Not sure if there is a simple solution to it, but I wasn't able to come up with a decent solution with the popover implementation.

**Comment by collinpreston** [comment]
@pdpinch, bootstrap recommends using a modal instead of building custom handling to get tab'ing and keyboard accessibility to work with popovers like this.  
[Reference](https://getbootstrap.com/docs/5.0/components/popovers/#usage)
> In short, simply adding interactive controls to a popover is likely to make these controls unreachable/unusable for keyboard users and users of assistive technologies, or at the very least make for an illogical overall focus order. In these cases, consider using a modal dialog instead.

Should we try our best to make the popover accessible to keyboard users or should we migrate to using a modal?
